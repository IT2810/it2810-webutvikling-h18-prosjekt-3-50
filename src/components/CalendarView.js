import React, { Component } from 'react'
import { Calendar, CalendarList, Agenda } from 'react-native-calendars'
import { View, Text } from 'native-base'

class CalendarView extends Component {
  getworkouts () {
  	return ''
  }

  addSession (date) {
  	console.log('Open view to add session on that date')
  }

  render () {
    const strength = {key: 'strength', color: 'red', selectedDotColor: 'blue'}
    const interval = {key: 'interval', color: 'blue', selectedDotColor: 'blue'}
    const calestethic = {key: 'calestethic', color: 'green'}

    const { navigate } = this.props.navigation

    return (
      <View>
        <Calendar
          markingType={'multi-dot'}
          onDayLongPress={(day) => { 
            navigate('CreateSession', {
              date: day
            })
          }}
        />
      </View>
    )
  }
}

export default CalendarView
