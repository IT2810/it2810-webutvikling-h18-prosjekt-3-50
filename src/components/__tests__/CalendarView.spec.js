/**
 * @jest-environment jsdom
 */

import React from 'react'
import { shallow } from 'enzyme'
import { CalendarView } from '../CalendarView'
import toJson from 'enzyme-to-json'
import { findByID } from '../../testUtils.js'

const initial_state_mock = require('../../assets/initial_state_mock.js')

describe('CalendarView', () => {
  let wrapper
  let navigateMock
  let navigation
  let dispatchSelectDateMock
  let sessionDates
  const sessions = initial_state_mock.sessions

  beforeEach(() => {
    sessionDates = []
    navigateMock = jest.fn()
    navigation = {navigate: navigateMock}
    dispatchSelectDateMock = jest.fn()
    dispatchCreateNewSessionMock = jest.fn()
    wrapper = shallow(<CalendarView
      navigation={navigation}
      selectDate={dispatchSelectDateMock}
      createNewSession={dispatchCreateNewSessionMock}
      sessionDates={sessionDates}
      sessions={sessions}
      />
    )
  })

  it('renders correctly', () => {
    expect(toJson(wrapper.dive())).toMatchSnapshot()
  })

  describe('addSession', () => {
    it('called when long press on day in calendar', () => {
      let addSessionMock = jest.fn()

      wrapper.instance().addSession = addSessionMock

      wrapper.instance().forceUpdate()

      let date = findByID(wrapper, 'calendar')
      date.props().onDayLongPress('2018-10-18')

      expect(addSessionMock.mock.calls.length).toBe(1)
    })

    it('calls selectDate and navigate with date as param', () => {
      const wrapper = shallow(<CalendarView navigation={navigation} selectDate={dispatchSelectDateMock} createNewSession={dispatchCreateNewSessionMock} sessions={sessions}/>)

      let date = findByID(wrapper, 'calendar')
      date.props().onDayLongPress('2018-10-18')

      expect(dispatchSelectDateMock.mock.calls.length).toBe(1)
      expect(navigateMock.mock.calls.length).toBe(1)
      expect(navigateMock.mock.calls[0][0]).toBe('CreateSession')
    })
  })

  describe('selectDate', () => {
    it('called when press on day in calendar', () => {
      let selectDateMock = jest.fn()

      wrapper.instance().selectDate = selectDateMock
      wrapper.instance().showToast = jest.fn()

      wrapper.instance().forceUpdate()

      let calendar = findByID(wrapper, 'calendar')
      calendar.props().onDayPress('2018-10-18')

      expect(selectDateMock.mock.calls.length).toBe(1)
    })

    it('setting date in state', () => {
      let dateString = '2018-10-18'
      let date = { dateString: dateString }
      let calendar = findByID(wrapper, 'calendar')
      calendar.props().onDayPress(date)

      expect(wrapper.state('date')).toBe(date)
    })

    it('add date to markedDates with select as true', () => {
      let dateString = '2018-10-18'
      let date = { dateString: dateString }
      let calendar = findByID(wrapper, 'calendar')
      calendar.props().onDayPress(date)

      let dateInMarkedDates = wrapper.state('markedDates')[date.dateString]
      expect(dateInMarkedDates.selected).toBe(true)
    })

    it('set selected date as marked if it is marked', () => {
      let dateString = '2018-10-18'
      let date = { dateString: dateString }
      let sessionDateObject = { [dateString]: { marked: true } }

      wrapper = shallow(<CalendarView
        navigation={navigation}
        selectDate={dispatchSelectDateMock}
        createNewSession={dispatchCreateNewSessionMock}
        sessionDates={sessionDateObject}
        sessions={sessions}
        />
      )
      let calendar = findByID(wrapper, 'calendar')
      calendar.props().onDayPress(date)

      let dateInMarkedDates = wrapper.state('markedDates')[dateString]
      expect(dateInMarkedDates.selected).toBe(true)
      expect(dateInMarkedDates.marked).toBe(true)
    })
  })
})
