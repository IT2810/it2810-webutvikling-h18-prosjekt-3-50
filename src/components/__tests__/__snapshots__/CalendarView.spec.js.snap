// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarView renders correctly 1`] = `
<ViewNB
  style={
    Array [
      Object {},
      Object {
        "marginBottom": "5%",
        "marginTop": "5%",
      },
    ]
  }
>
  <Calendar
    markedDates={Object {}}
    onDayLongPress={[Function]}
    onDayPress={[Function]}
    testID="calendar"
  />
</ViewNB>
`;
